### PR TITLE
chore(tile-debug): split logic between multiple files

### PR DESCRIPTION
**Pull request by dryabinin-actionengine**
1. Split tile-debug.ts to multiple helper files.
2. Add types to the files.

### CODE REVIEW

File: src/utils/geometry-vs-texture-metrics.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import type { Tile3D } from "@loaders.gl/tiles";
```
Comment (belom88) [inline src/utils/geometry-vs-texture-metrics.ts:1]: I suggest moving this file in `attributes-validation` folder.

File: src/utils/lod-validation.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import type { Tile3D } from "@loaders.gl/tiles";
```
Comment (belom88) [inline src/utils/lod-validation.ts:1]: I suggest moving this file in `tile-validation` folder.

File: src/utils/tile-validation-data.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import type { TileValidationData } from "./types";
```
Comment (belom88) [inline src/utils/tile-validation-data.ts:1]: I suggest moving this file in `tile-validation` folder.

File: src/utils/triangles-calculation.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import type { MeshAttribute } from '@loaders.gl/schema';
```
Comment (belom88) [inline src/utils/triangles-calculation.ts:1]: I suggest moving this file in `attributes-validation` folder.

File: src/utils/bounding-volume-from-tile.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import type { ObbData } from './types';
```
Comment (belom88) [inline src/utils/bounding-volume-from-tile.ts:1]: I suggest joining this file and `get-volume-type` in `bounding-volume.ts`

File: src/utils/bounding-volume-from-tile.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import { OrientedBoundingBox, BoundingSphere } from "@math.gl/culling";
+import { OBB, MBS } from "../constants/bounding-volumes";
+
+/**
+ * Generates needed bounding volume from tile bounding volume
+ * @param tile
+ * @param boundingType
+ */
+export const createBoundingVolumeFromTile = (tile: Tile3D, boundingType: string): BoundingSphere | OrientedBoundingBox => {
+  switch (boundingType) {
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/utils/bounding-volume-from-tile.ts]: Done

File: src/utils/convert-positions-to-vectors.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import { Vector3 } from "@math.gl/core";
+/**
+ * Create array of posisitons where each vertex is vector
+ * @param positions
+ */
+export const convertPositionsToVectors = (positions: Float32Array): Vector3[] => {
+  const result: Vector3[] = [];
+
+  for (let i = 0; i < positions.length; i += 3) {
+    const positionVector = new Vector3(
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/utils/convert-positions-to-vectors.ts]: Done

File: src/utils/tile-validation-data.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  ) {
+    throw new Error("Validator - There are no positions in tile");
+  }
+
+  const boundingType = getBoundingType(tile);
+  const positions = tile.content.attributes.positions.value;
+
+  const boundingVolume = createBoundingVolumeFromTile(tile, boundingType);
+  return { positions, boundingType, boundingVolume };
+};
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/utils/tile-validation-data.ts]: Done

File: src/utils/triangles-calculation.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  for (let i = 0; i < 3; i++) {
+    // @ts-expect-error - This expression is not constructable. Type 'Function' has no construct signatures.
+    const typedArray = new attribute.value.constructor(3);
+    const subarray = attribute.value.subarray(
+      (offset + i) * attribute.size,
+      (offset + i) * attribute.size + attribute.size
+    );
+    typedArray.set(subarray);
+
+    geometryVertices.push(new Vector3(typedArray));
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/utils/triangles-calculation.ts]: Done

File: src/utils/triangles-calculation.ts [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  }
+  return geometryVertices;
+};
+
+/**
+ * Calculates triangles area based on vertices
+ * @param vertices
+ */
+export const getTriangleArea = (vertices: Vector3[]): number => {
+  const edge1 = new Vector3(
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/utils/triangles-calculation.ts]: Done

File: src/utils/triangles-calculation.ts [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    vertices[0].x,
+    vertices[0].y,
+    vertices[0].z
+  ).subtract(vertices[1]);
+  const edge2 = new Vector3(
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/utils/triangles-calculation.ts]: Done

File: src/utils/bounding-volume-from-tile.ts (hunk 5 above)
Comment (dryabinin-actionengine) [reply src/utils/bounding-volume-from-tile.ts]: Done

File: src/utils/bounding-volume-validation.ts [hunk 12]
```diff
@@ -0,0 +1,114 @@
+import type { Tile3D } from "@loaders.gl/tiles";
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/utils/bounding-volume-validation.ts]: Done

File: src/utils/lod-validation.ts (hunk 2 above)
Comment (dryabinin-actionengine) [reply src/utils/lod-validation.ts]: Done

File: src/utils/tile-validation-data.ts (hunk 3 above)
Comment (dryabinin-actionengine) [reply src/utils/tile-validation-data.ts]: Done

File: src/utils/triangles-calculation.ts (hunk 4 above)
Comment (dryabinin-actionengine) [reply src/utils/triangles-calculation.ts]: Done

File: src/utils/geometry-vs-texture-metrics.ts (hunk 1 above)
Comment (dryabinin-actionengine) [reply src/utils/geometry-vs-texture-metrics.ts]: done

**Review by belom88 — APPROVED** [review]

**Pull request merged**